cpp-client/build-dependencies.sh

```diff
@@ -5,12 +5,30 @@
 #
 
 #
-# Tested on Ubuntu 20.04
+# Tested on Ubuntu 22.04 and Fedora 28.
+#
+# On Ubuntu, the following packages need to be installed:
+#
+# $ sudo apt -y install git g++ cmake make build-essential zlib1g-dev libssl-dev pkg-config
+#
+# On Fedora, the following packages need to be installed:
+#
+# $ sudo dnf -y groupinstall 'Development Tools'
+# $ sudo dnf -y install cmake gcc-c++ openssl-devel
 #
 
 # Fail on first error; echo each command before executing.
 set -euo pipefail
 
+if [ -f /etc/redhat-release ]; then
+  fedora=yes
```

Comment:
Does this exist on _Fedora_? It seems like there should be a more appropriately named file for _Fedora_?

Reply:
```
cfs@tenten 11:59:28 ~
$ docker run -it fedora:28 /bin/bash
[root@7ec7e813ba3b /]# ls -l /etc/redhat-release
lrwxrwxrwx 1 root root 14 Oct  4  2018 /etc/redhat-release -> fedora-release
```

This is better.  It means the detection will also work in RHEL (although we don't guarantee the build will work there, because I don't want to have to test multiple builds).

Reply:
And in case you were wondering:
```
cfs@tenten 21:19:52 ~/dh/oss1/deephaven-base-images
$ docker run -it fedora:38 /bin/bash
[root@f401cee151d1 /]# ls -l /etc/redhat-release
lrwxrwxrwx 1 root root 14 May 30 00:00 /etc/redhat-release -> fedora-release
```